tests/test_trial.py

```diff
@@ -425,3 +425,9 @@ def test_trial_report():
 
     with pytest.raises(TypeError):
         trial.report([1, 2, 3])  # type: ignore
+
+    with pytest.raises(TypeError):
+        trial.report('foo', -1)  # type: ignore
+
+    with pytest.raises(ValueError):
+        trial.report(1.23, -1)  # type: ignore
```

Comment:
[nits] I'm not sure but it seems that `# type: ignore` can be removed.

Reply:
This is optional.
```suggestion
        trial.report(1.23, -1)
```

```python
    def report(self, value, step=None):
        # type: (float, Optional[int]) -> None
```

Reply:
Ah, thanks you for the catch, it's obviously a `ValueError` and not a `TypeError`. Fixed.